### ISSUE TITLE
Kap is extremely slow to show capture selection (10 seconds)

### ISSUE THREAD

**Issue by nicmar** [wulkano/Kap#1125]
**macOS version:** 10.12.1
**Kap version:** Version 3.5.4 (3.5.4.1810)

#### Steps to reproduce
Pressing Kap button in the menubar

#### Current behaviour
Takes 8-10 seconds before i can select an area to record. Also right click "About Kap" doesn't give any information or feedback.

#### Expected behaviour
In the earlier versions, this was extrememly fast, like less than 1 second, but it has been slower in later versions, like 3-4 seconds. It was still acceptable.

#### Workaround
 No workaround, just have to wait.

It might have been worse since upgrading to Monterey but I'm not sure.

**Comment by malandles** [comment]
Same behavior on macOS Monterey 12.4 with Kap version 3.5.4 (3.5.4.1810).

**Comment by sveggiani** [comment]
Noticing this for a time now. It have made it almost unusable for me :/

**Comment by carlosperate** [comment]
Having the same issue in an Intel Macbook Pro with macOS Monterey 12.1.

I've tried going back to each version from the releases pages until one of them worked well. In this case I had to go back to v3.4.1.

v3.4.2 didn't work (likely the signing issues indicated in the release notes) and v3.5.0 already shows this problem. Looking at the diff, is fairly small, so maybe we can try manually reverting these changes to see if it makes a difference: https://github.com/wulkano/Kap/compare/v3.4.1...v3.5.0

**Comment by wkoutre** [comment]
Same issues here, macOS 12.4 w/ multiple monitors.

Downgrading to 3.4.1 as @carlosperate suggested has worked for me.

**Comment by ajwhite** [comment]
Can confirm `3.4.1` resolved the issue for me as well.

**Comment by ajwhite** [comment]
@carlosperate this contribution stands out to me in that diff, https://github.com/wulkano/Kap/pull/1051 -- upgrading electron & supporting different build target architectures

@codetheweb @sindresorhus can you see that impacting this behavior?

**Comment by ajwhite** [comment]
Cross referencing similar thread https://github.com/wulkano/Kap/issues/1088

**Comment by codetheweb** [comment]
Hmm, something must have changed in the new Electron version.

Might be related: https://github.com/desktop/desktop/issues/12978

**Comment by literalpie** [comment]
Any idea if going to a newer version of electron would fix this? The latest is v19.0, and Kap is on 12.x. Going to new electron would also move this from Node 14.x to 16.x.

**Comment by codetheweb** [comment]
Possibly, but upgrading versions would require quite a few changes.

**Comment by nicmar** [comment]
Yeah 3.4.1 seems to have worked. But how do I prevent it from autoupdating? I don't see any option like that, but I assume it did cause I never updated it manually.

**Comment by nicmar** [comment]
I keep getting this error, which I guess is a good sign that it can't update :P

> Unhandled Promise Rejection
> Error: Could not get code signature for running application
> at Server.<anonymous> (/Applications/Kap.app/Contents/Resources/app.asar/node_modules/electron-updater/out/MacUpdater.js:105:44)

**Comment by nicmar** [comment]
To author: Please have auto updates as an option as well, to prevent future issues like this. Thanks :)

**Comment by Vignesh-Au** [comment]
Same here. 10 seconds after clicking kap icon.

**Comment by xinatcg** [comment]
same 3.5.5, hard to use

**Comment by pkoziol** [comment]
Same here after upgrading from Catalina to Monterey (12.5.1). Probably also upgraded Kap currently on 3.5.5.1825

Kap 3.4.1 is ineed much faster.

**Comment by jesshart** [comment]
For anyone looking to download older versions, I went ahead and did this:
1. Go to https://github.com/wulkano/Kap/releases
2. Find the release you want
3. Click `Assets` and find your corresponding download file.
4. Download the file.
5. Run the download file.

All of the downloads `<3.5.5` had a pop-up warning on my Mac. Proceed at your own risk by clicking the app while holding <kbd>control</kbd>. Follow the prompt and choose to open or not.

**Comment by okyr-ux** [comment]
**macOS version:**    Monterey 12.6.1 (21G217)
**Kap version:**      3.6.0
MacBook Pro (Intel) + external monitor.

I've just updated from 2.2.0 to 3.6.0 and at first I didn't even realise what's going on: after clicking the menubar icon nothing happens, then there's a rainbow ball, then the whole screen becomes dark (I already get that it's a feature), the laptop starts to take off, according to the fans sound.

Version 2.2.0 was also not very responsive, but it was definitely more user-friendly.

I'll go try the 3.4.1 now.

**Comment by sveggiani** [comment]
macOS version: Monterey 12.6.1 (21G217)
Kap version: 3.6.0
MacBook Pro (Apple M1 Pro) + external monitor.

For me, v3.6.0 is not slow anymore. Everything looks good again 🙌

**Comment by wkoutre** [comment]
![image](https://user-images.githubusercontent.com/13321210/199824686-7c28ffd2-66ab-4585-8536-4119b2404487.png)

Kap 3.6.0 is _better_, but still takes 7-10 seconds from initializing a session to seeing the GUI appear

**Comment by timothyis** [comment]
Trying to remedy this by updating Electron, by the way. It's extremely difficult, unfortunately.

**Comment by wojtiku** [comment]
Some for me, it has been extremely slow in the past months. Trying to downgrade as others suggest.

**Comment by WilCF** [comment]
Same for me, unusable current version. Beachball of death.

**Comment by zremboldt** [comment]
Installed the latest version from the get kap homepage and am having this issue too.
Also tried downgrading as suggested with no luck.

**Comment by JeffKko** [comment]
v3.4.1 works for me!

**Comment by jt3k** [comment]
Version 3.6.0 (3.6.0.1846)
iMac Pro 3,2 GHz 8‑core Intel Xeon W, Radeon Pro Vega 56 8 GB, 32 GB 2666 MHz DDR4

still freezes on startup and before the recording interface appears.
I see a beach ball instead of a mouse cursor

**Comment by ischenxin** [comment]
Is there any progress on this issue? I'm still experiencing the same problem in the latest version 3.6.0. Every time I start recording, it's very slow, and it's causing a lot of inconvenience for me.

**Comment by jdk2pq** [comment]
@timothyis [This branch](https://github.com/wulkano/Kap/tree/update/replace-internal-remote-usage ) is working great for me on an M1 Macbook Pro, and Kap's back to being speedy again. I built it following the directions from [contributing.md](https://github.com/wulkano/Kap/blob/update/replace-internal-remote-usage/contributing.md). Unfortunately, it doesn't look like the link to the dev builds is working anymore: https://kap-artifacts.vercel.app/main

**Comment by AMerkuri** [comment]
3.6.0 extremely slow on MacBook Pro 16-inch, 2019 2,6 GHz 6-Core Intel Core i7 macbook. Ventura 13.5.2 (22G91)

**Comment by CleverCoder** [comment]
> @timothyis [This branch](https://github.com/wulkano/Kap/tree/update/replace-internal-remote-usage) is working great for me on an M1 Macbook Pro, and Kap's back to being speedy again. I built it following the directions from [contributing.md](https://github.com/wulkano/Kap/blob/update/replace-internal-remote-usage/contributing.md). Unfortunately, it doesn't look like the link to the dev builds is working anymore: https://kap-artifacts.vercel.app/main

I have yet to pull/build locally, but if this addresses the issue, can it be PR'd, merged in?

**Comment by michaeljelly** [comment]
I've fixed this, it's working nice and quick for me now, it was a few changes around not auto-closing windows all the time, and not triggering full rebuilds of context menus on every click.

If someone wants to clean it up to try to merge it, feel free! Kap is fast again now.
https://github.com/wulkano/Kap/pull/1208

**Comment by gruns** [comment]
same problem with v3.6.0 on m1 MBA, os x 15

**Comment by okyr-ux** [comment]
@gruns @michaeljelly please use the [v.3.4.1](https://github.com/wulkano/Kap/releases/tag/v3.4.1).

UPD: …or try another older version. I found that v.3.4.1. works great on my old MBP with macOS Monterey (12.7.6).
